Makefile

```diff
@@ -1,60 +1,23 @@
-GTEST_DIR ?= external/googletest/googletest
-CXX ?= clang++
-CXXFLAGS += -std=c++14 -pedantic -Wall -Wextra -Iinclude/
-CC ?= clang
-CFLAGS += -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter -Iinclude/
-AR ?= ar
-RANLIB ?= ranlib
-XCODEBUILD ?= xcodebuild
+build := build
+cmake := cmake
+make := make
 
-SOURCES = $(shell find src -name '*.cpp')
-OBJECTS = $(SOURCES:.cpp=.o)
-LIBRARY = libArbiter.a
+all:
+	mkdir -p $(build)
+	$(cmake) -B$(build) -H. -G "Unix Makefiles" -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE
+	$(make) -C $(build)
 
-TEST_SOURCES = $(shell find test -name '*.cpp') $(GTEST_DIR)/src/gtest-all.cc $(GTEST_DIR)/src/gtest_main.cc
-TEST_RUNNER = test/main
-TEST_INCLUDES = -isystem $(GTEST_DIR)/include -I$(GTEST_DIR) -Isrc/
-
-EXAMPLES = examples/library_folders/library_folders
-EXAMPLE_LIBRARY_FOLDERS = $(shell find examples/library_folders -name '*.c')
-EXAMPLE_LIBRARY_FOLDERS_OBJECTS = $(EXAMPLE_LIBRARY_FOLDERS:.c=.o)
-
-.PHONY: bindings/swift check docs
-
-all: build
+check: all
+	set -e; for test in `find $(build) -name "*Test" -type f -perm +111`; do echo; echo "$$test"; ./$$test; done
```

Comment:
Would you mind factoring this out into its own shell script, a la `script/cibuild`?